src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1984,6 +1999,7 @@ public Status do_PUT_Document(Database _db, String docID, String _attachmentName
                 throw new CouchbaseLiteException(Status.BAD_REQUEST);
             }
             List<String> history = Database.parseCouchDBRevisionHistory(body.getProperties());
+            // forceInsert uses transaction internally, not necessary to apply synchronized
             db.forceInsert(rev, history, source);
```

Comment:
[forceInsert()](https://github.com/couchbase/couchbase-lite-java-core/blob/a35361bc47ecb6c55849694deece22fa4aaf7c62/src/main/java/com/couchbase/lite/store/SQLiteStore.java#L1545) uses transaction in Store layer

https://github.com/couchbase/couchbase-lite-java-core/blob/a35361bc47ecb6c55849694deece22fa4aaf7c62/src/main/java/com/couchbase/lite/store/SQLiteStore.java#L1545